src/hooks/useRefValue.ts

```diff
@@ -0,0 +1,11 @@
+import { useRef } from 'react'
+
+// @see https://react.dev/reference/react/useRef#avoiding-recreating-the-ref-contents
+export function useRefValue<T>(value: () => T): T {
+  const ref = useRef<T>()
+
+  if (!ref.current)
```

Comment:
If this value is a function that may returns `undefined`/`null`, is it necessary to declare its type annotation as `() => T | undefined | null` ?